cfbs/commands.py

```diff
@@ -534,24 +517,55 @@ def update_command(to_update):
             if key not in index_info or module[key] == index_info[key]:
                 continue
             if key == "steps":
+                # inputs may cause a specific modification in build steps
+                input_step = "json ./%s/def.json def.json" % module["name"]
+                if module["steps"] == index_info["steps"] + [input_step]:
+                    continue
                 # same commit => user modifications, don't revert them
                 if commit_differs:
                     ans = prompt_user(
                         config.non_interactive,
-                        "Module %s has different build steps now\n" % module["name"]
+                        "Module '%s' has different build steps now\n" % module["name"]
                         + "old steps: %s\n" % module["steps"]
                         + "new steps: %s\n" % index_info["steps"]
                         + "Do you want to use the new build steps?",
                         choices=YES_NO_CHOICES,
                         default="yes",
                     )
-                    if ans.lower() in ["y", "yes"]:
+                    if ans.lower() in ("y", "yes"):
                         module["steps"] = index_info["steps"]
+                        changes_made = True
                     else:
                         print(
                             "Please make sure the old build steps work"
                             + " with the new version of the module"
                         )
+            if key == "input":
+                if commit_differs:
+                    ans = prompt_user(
+                        config.non_interactive,
+                        "Module '%s' takes different input now. " % update.name
+                        + "Do you want to use new inputs?",
+                        choices=YES_NO_CHOICES,
+                        default="yes",
+                    )
+                    if ans.lower() in ("y", "yes"):
+                        module["input"] = index_info["input"]
+                        changes_made = True
+
+                        ans = prompt_user(
+                            config.non_interactive,
+                            "Do you want to add input for module '%s' right away?",
+                            choices=YES_NO_CHOICES,
+                            default="no",
+                        )
```

Comment:
This second prompt is weird. The user just told us they wanted to use new inputs. I think the two prompts should be combined into a single one giving the user an option to use the defaults.